client/src/components/formHelpers/index.tsx

```diff
@@ -1,27 +1,44 @@
 import normalizeUrl from 'normalize-url';
+import { ValidatedValues } from '../../templates/Challenges/projects/solution-form';
+import { FormOptions } from './form';
 import {
   localhostValidator,
   editorValidator,
   composeValidators,
   fCCValidator,
   httpValidator
-} from './FormValidators';
+} from './form-validators';
 
 export { default as BlockSaveButton } from './block-save-button';
 export { default as BlockSaveWrapper } from './block-save-wrapper';
-export { default as Form } from './Form.js';
-export { default as FormFields } from './FormFields.js';
+export { default as Form } from './form';
+export { default as FormFields } from './form-fields';
 
 const normalizeOptions = {
   stripWWW: false
 };
 
-export function formatUrlValues(values, options) {
+type validatedValuesType = {
+  errors: string[];
+  invalidValues: string[];
+  values: Record<string, unknown>;
+};
+
+export function formatUrlValues(
+  values: ValidatedValues,
+  options: FormOptions
+): validatedValuesType {
   const { isEditorLinkAllowed, isLocalLinkAllowed, types } = options;
-  const validatedValues = { values: {}, errors: [], invalidValues: [] };
+  const validatedValues: validatedValuesType = {
+    values: {},
+    errors: [],
+    invalidValues: []
+  };
   const urlValues = Object.keys(values).reduce((result, key) => {
-    let value = values[key];
-    const nullOrWarning = composeValidators(
+    // eslint-disable-next-line
+    //@ts-ignore
+    let value: string = values[key] as string;
+    const nullOrWarning: string = composeValidators(
```

Comment:
Unfortunately, client/src/templates/Challenges/projects/solution-form.tsx is misleading, the `ValidatedValues` type is incorrect.  To fix this, I would infer the type from `composeValidators` and correct `solution-form.tsx`'s typing to use that.

Reply:
@ojeytonwilliams I'm not sure what do you mean by referring the validatedValues type from ComposeValidators.

Reply:
So, the `ValidatedValues` type in client/src/templates/Challenges/projects/solution-form.tsx is not correct.  What you can do is start from client/src/components/formHelpers/form-validators.tsx and set the validators return type to `JSX.Element | null`, then see what other changes that requires.